test/local_pow_test.py

```diff
@@ -0,0 +1,197 @@
+        # ccurl_bundle is only needed to differentiate between response
+        # from mocked pow and MockAdapter in some test cases.
+        self.ccurl_bundle = [TransactionTrytes(self.trytes1)]
+        self.trunk = TransactionHash(self.trunk)
+        self.branch = TransactionHash(self.branch)
+    
+    def test_backward_compatibility(self):
+        """
+        Test that the local_pow feature is backward compatible.
+        That is, if `local_pow` argument is omitted, it takes no
+        effect and the pow extension package is not called.
+        """
+        with patch('pow.ccurl_interface.attach_to_tangle',
+                   MagicMock(return_value=self.ccurl_bundle)) as mocked_ccurl:
+            self.adapter = MockAdapter()
+            self.adapter.seed_response('attachToTangle',{
+                'trytes': self.bundle,
+            })
+            # No `local_pow` argument is passed to the api!
+            api = Iota(self.adapter)
+            result = api.attach_to_tangle(
+                self.trunk,
+                self.branch,
+                self.bundle,
+                self.mwm)
+            # Ccurl interface was not called
+            self.assertFalse(mocked_ccurl.called)
+            # Result is the one returned by MockAdapter
+            self.assertEqual(result['trytes'], self.bundle)
+            # And not by mocked pow pkg
+            self.assertNotEqual(result['trytes'], self.ccurl_bundle)
+
+    def test_http_adapter(self):
+        """
+        Test if local_pow feature works with HttpAdapter.
+        """
+        # Note that we need correct return value to pass the
+        # repsonse filter.
+        with patch('pow.ccurl_interface.attach_to_tangle',
+                    MagicMock(return_value=self.bundle)) as mocked_ccurl:
+            api = Iota(HttpAdapter('http://localhost:14265/'),local_pow=True)
+            result = api.attach_to_tangle(
+                self.trunk,
+                self.branch,
+                self.bundle,
+                self.mwm)
+            self.assertTrue(mocked_ccurl.called)
+            self.assertEqual(result['trytes'], self.bundle)
+
+    def test_mock_adapter(self):
+        """
+        Test if local_pow feature works with MockAdapter.
+        """
+        # Note that we need correct return value to pass the
+        # repsonse filter.
+        with patch('pow.ccurl_interface.attach_to_tangle',
+                    MagicMock(return_value=self.bundle)) as mocked_ccurl:
+            api = Iota(MockAdapter(),local_pow=True)
+            result = api.attach_to_tangle(
+                self.trunk,
+                self.branch,
+                self.bundle,
+                self.mwm)
+            self.assertTrue(mocked_ccurl.called)
+            self.assertEqual(result['trytes'], self.bundle)
+
+    def test_sandbox_adapter(self):
+        """
+        Test if local_pow feature works with SandboxAdapter.
+        """
```

Comment:
💭  Out of curiosity, is the sandbox node still running?  Is it time to remove `SandboxAdapter`?

Reply:
I think I need to be enlightened on this topic. What's the story with "the sandbox node"?

Reply:
I think it was a couple of years ago now, the IF stood up a special node that would allow clients to outsource the PoW for `attachToTangle`.  It was an asynchronous process; the client would send the request, and it would get a job ID back.  The client would then need to poll the node until the PoW had been finished, and then it could retrieve the trytes ready to broadcast.

Announcement here:
https://blog.iota.org/iota-sandbox-developer-environment-c70c8206fe5f

The source code for the sandbox node itself is here:
https://github.com/iotaledger/sandbox